src/mbgl/map/map.cpp

```diff
@@ -794,6 +798,16 @@ style::Layer* Map::getLayer(const std::string& layerID) {
     return nullptr;
 }
 
+void Map::setLayers(std::vector<std::unique_ptr<style::Layer>> layers) {
+    impl->view.activate();
```

Comment:
Needs an `if (!impl->style)` guard and `impl->styleMutated = true;` like the other mutators.